prompt/prompt-executor/prompt-executor-clients/prompt-executor-bedrock-client/src/jvmMain/kotlin/ai/koog/prompt/executor/clients/bedrock/BedrockLLMClient.jvm.kt

```diff
@@ -0,0 +1,645 @@
+                    response.body?.collect { event: ResponseStream ->
+                        val chunkBytes = event.asChunk()?.bytes
+                        if (chunkBytes != null) {
+                            val chunkJsonString = chunkBytes.decodeToString()
+                            send(chunkJsonString)
+                            logger.trace { "Bedrock Stream Chunk for model ${model.id} (JVM): $chunkJsonString" }
+                        } else {
+                            logger.warn { "Received null chunk bytes in stream for model ${model.id} (JVM)" }
+                        }
+                    }
+                }
+            } catch (e: Exception) {
+                logger.error(e) { "Error in Bedrock streaming for model ${model.id} (JVM)" }
+                close(e)
+            }
+        }.map { chunkJsonString ->
+            try {
+                if (chunkJsonString.isBlank()) return@map ""
+
+                when {
+                    model.id.startsWith("anthropic.claude") -> parseAnthropicStreamChunk(chunkJsonString)
+                    model.id.startsWith("amazon.titan") -> parseTitanStreamChunk(chunkJsonString)
+                    model.id.startsWith("meta.llama") -> parseLlamaStreamChunk(chunkJsonString)
+                    else -> ""
+                }
+            } catch (e: Exception) {
+                logger.warn(e) { "Failed to parse Bedrock stream chunk (JVM): $chunkJsonString" }
+                ""
+            }
+        }
+    }
+
+    // Anthropic Claude specific methods
+    private fun createAnthropicRequest(prompt: Prompt, model: LLModel, tools: List<ToolDescriptor>): JsonObject {
+        val messages = mutableListOf<JsonObject>()
+        val systemMessages = mutableListOf<String>()
+
+        prompt.messages.forEach { msg ->
+            when (msg) {
+                is Message.System -> systemMessages.add(msg.content)
+                is Message.User -> {
+                    // Check for image content if present
+                    if (msg.mediaContent.isNotEmpty()) {
+                        require(model.capabilities.contains(LLMCapability.Vision.Image)) {
+                            "Model ${model.id} does not support image input"
+                        }
+
+                        // Handle multimodal content
+                        val contentParts = buildJsonArray {
+                            if (msg.content.isNotEmpty()) {
+                                add(buildJsonObject {
+                                    put("type", "text")
+                                    put("text", msg.content)
+                                })
+                            }
+
+                            msg.mediaContent.forEach { media ->
+                                when (media) {
+                                    is MediaContent.Image -> {
+                                        val imageData = if (media.isUrl()) {
```

Comment:
It seems that `imageData` variable is unused after creating. Could you please clarify if it is used anywhere below?